lib/ws.js

```diff
@@ -107,10 +139,14 @@ class BinanceWS {
     }
 
     onUserData(binanceRest, eventHandler, interval = 60000) {
+        this._clearUserDataInterval();
         return binanceRest.startUserDataStream().then(response => {
-            setInterval(() => {
-                binanceRest.keepAliveUserDataStream(response);
-            }, interval);
+            this.userDataRefreshIntervalId = setInterval(
+                () => this._sendUserDataKeepAlive(binanceRest, response),
+                interval
+            );
+            this.userDataRefreshIntervalId.failCount = 0;
```

Comment:
Does this actually work?  Afaik `setInterval` returns a number, and then you're trying to set a `failCount` property on the number.   A quick test in my browser console seems to indicate it doesn't work, but it might be different in node.js.  Either way, I think it should be changed as it's kind of confusing.  I would either use an object to store both numbers or just create an additional private property on the class instance:

```javascript

this._userDataRefresh = {}; // Perhaps in the constructor
this._userDataRefresh.intervaId = setInterval();
this._userDataRefresh.failCount = 0;
}

// or

this._userDataRefreshIntervalId = setInterval()
this._userDataRefreshFailCount = 0;
```

Reply:
I didn't think this would be different in the browser. In node, setInterval seems to return an object, but agree this is a bit lazy & confusing. I'll consolidate into an object with two defined properties.